packages/teams-js/src/public/secondaryBrowser.ts

```diff
@@ -0,0 +1,50 @@
+import { sendAndHandleSdkError as sendAndHandleError } from '../internal/communication';
+import { ensureInitialized, isHostClientMobile } from '../internal/internalAPIs';
+import { isValidHttpsURL } from '../internal/utils';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to power up the in-app browser experiences in the Host App.
+ * For eg. opening a URL in the Host App inside a browser
+ *
+ * @beta
+ */
+export namespace secondaryBrowser {
+  /**
+   * Open a URL in the secondary browser aka in-app browser
+   *
+   * @param url Url to open in the browser
+   * @returns Promise that resolve to true if the URL successfully opens in the secondaryBrowser
+   * or throws an error {@link SdkError} incase of failure before starting navigation
+   *
+   * Pls note that any error that happens after starting navigation is delegated on external browsers to handle in their generic way
+   * @beta
+   */
+  export function open(url: URL): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content);
+    if (!isSupported() || !isHostClientMobile()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    if (!url || !isValidHttpsURL(url)) {
```

Comment:
nit: maybe this check can be inside isValidHttpsURL?

Reply:
just to be more clear, I meant to say !url can be inside the function isValidHttpsURL :)